### PR TITLE
Fix bugs when inputing wrong format which will return server error

### PR DESCRIPTION
**Pull request by simozhan**
JIRA: PDC-1560

### CODE REVIEW

File: pdc/apps/compose/views.py [hunk 1]
```diff
@@ -993,7 +997,7 @@ def _update_parameters_acceptable(self, in_data, layer):
         if layer == 1:
             result = isinstance(in_data, list)
         elif layer > 1:
-            if not isinstance(in_data, dict) and len(in_data) != 1:
+            if not isinstance(in_data, dict):
```
Comment (ycheng-aa) [inline pdc/apps/compose/views.py]: Should we allow dict length greater than 1?

**Comment by lubomir** [comment]
LGTM. The reason why the RPM mapping end-point was not very robust with regards to bad input was that when it was written, it was not clear whether it actually satisfies RCM requirements. As far as I know, there has been no use of it yet, so we still don't know.

File: pdc/apps/compose/views.py (hunk 1 above)
Comment (simozhan) [inline pdc/apps/compose/views.py]: Empty dict shouldn't be allowed, will change it.

**Comment by simozhan** [comment]
@ycheng-aa @lubomir Currently i changed it to not allow empty and duplicated dicts as input. If future they have further requirements when using this api, let's change it. Does it make sense?

**Comment by ycheng-aa** [comment]
:+1:

**Comment by lubomir** [comment]
@simozhan Makes perfect sense :+1:

**Pull request merged**